### PR TITLE
Rename data partition key to general purpose

### PR DESCRIPTION
**Pull request by Sindica**

### CODE REVIEW

File: docs/design-proposals/apiserver/Data-Partition-Design.md [hunk 1]
```diff
@@ -126,17 +126,17 @@ type DataPartitionConfig struct {
...
 	metav1.ObjectMeta
 
-	// Start tenant is inclusive
-	StartTenant string
+	// Range start is inclusive
+	RangeStart string
 
 	// Whether this is an open end start
-	IsStartTenantValid bool
+	IsRangeStartValid bool
```
Comment (jshaofuturewei) [inline docs/design-proposals/apiserver/Data-Partition-Design.md:133]: Is IsRangeStartValid related to inclusive, like true = inclusive, false = exclusive?

Comment (Sindica) [reply docs/design-proposals/apiserver/Data-Partition-Design.md:133]: true/false is used as limited or unlimted

Comment (jshaofuturewei) [reply docs/design-proposals/apiserver/Data-Partition-Design.md:133]: Do you mean ∞? 
If it is false(unlimited), the RangeStart/End will be meaningless, right?
There is a partition data equals statement  in staging/src/k8s.io/apiserver/pkg/storage/datapartition/datapartitionmanager.go made me a little confusing, if 
 a  and b have  both IsRangeStartValid and IsRangeEndValid false, they will be equal to each other no matter that they have different RangeStart/End value?

Comment (Sindica) [reply docs/design-proposals/apiserver/Data-Partition-Design.md:133]: yes. If IsRangeStartValid == false, RangeStart will be ignored.

Comment (jshaofuturewei) [reply docs/design-proposals/apiserver/Data-Partition-Design.md:133]: Thanks.

**Pull request merged**